### ISSUE TITLE
MSDOS 7.10 crashing on startup

### ISSUE THREAD

**Issue by andrewbird** [stsp/dosemu2#895]
**Describe the bug**
Crash bug on dosemu startup

**To Reproduce**
See the test case

**Attach the binaries**
[msdos710-crash.tar.gz](https://github.com/stsp/dosemu2/files/3487666/msdos710-crash.tar.gz)

**Attach the log**
[test.log](https://github.com/stsp/dosemu2/files/3487667/test.log)

**Comment by stsp** [comment]
Great finding, thanks.

**Comment by andrewbird** [comment]
Confirmed all booting again.
~~~
Test FR-DOS-1.00 FAT12 vfs directory ... ok
Test FR-DOS-1.10 FAT12 vfs directory ... ok
Test FR-DOS-1.20 FAT12 vfs directory ... ok
Test MS-DOS-3.10 FAT12 vfs directory ... ok
Test MS-DOS-3.20 FAT12 vfs directory ... ok
Test MS-DOS-3.21 FAT12 vfs directory ... ok
Test MS-DOS-3.30-Nec FAT12 vfs directory ... ok
Test MS-DOS-3.30 FAT12 vfs directory ... ok
Test MS-DOS-3.31 FAT12 vfs directory ... ok
Test MS-DOS-4.01 FAT12 vfs directory ... ok
Test MS-DOS-5.00 FAT12 vfs directory ... ok
Test MS-DOS-6.00 FAT12 vfs directory ... ok
Test MS-DOS-6.20 FAT12 vfs directory ... ok
Test MS-DOS-6.21 FAT12 vfs directory ... ok
Test MS-DOS-6.22 FAT12 vfs directory ... ok
Test MS-DOS-7.00 FAT12 vfs directory ... ok
Test MS-DOS-7.10 FAT12 vfs directory ... ok
Test PC-DOS-3.00-Compaq FAT12 vfs directory ... ok
Test PC-DOS-3.00 FAT12 vfs directory ... ok
Test PC-DOS-3.10-850307 FAT12 vfs directory ... ok
Test PC-DOS-3.10-850422 FAT12 vfs directory ... ok
Test PC-DOS-3.10-Compaq FAT12 vfs directory ... ok
Test PC-DOS-3.20-851230 FAT12 vfs directory ... ok
Test PC-DOS-3.20-860221 FAT12 vfs directory ... ok
Test PC-DOS-3.30 FAT12 vfs directory ... ok
Test PC-DOS-3.31-Compaq FAT12 vfs directory ... ok
Test PC-DOS-4.00 FAT12 vfs directory ... ok
Test PC-DOS-4.01 FAT12 vfs directory ... ok
Test PC-DOS-5.00 FAT12 vfs directory ... ok
Test PC-DOS-5.02 FAT12 vfs directory ... ok
Test PC-DOS-6.10 FAT12 vfs directory ... ok
Test PC-DOS-6.30 FAT12 vfs directory ... ok
Test PC-DOS-7.00 FAT12 vfs directory ... ok
Test PC-DOS-7.10 FAT12 vfs directory ... ok
Test PC-DOS-7.2K FAT12 vfs directory ... ok
Test PP-DOS-GIT FAT12 vfs directory ... ok
~~~

**Comment by stsp** [comment]
I wish all bugs to be that simple.
I expected something tricky with new interrupt
handling...

**Comment by ecm-pushbx** [comment]
I think you need to add the "load_offs = -ofs;" line to the RxDOS.2 boot protocol too; it is to load at 00700h but entered at 70h:400h.

**Comment by stsp** [comment]
Thanks!

**Comment by ecm-pushbx** [comment]
Btw, why did you remove the too large size checks for FreeDOS and RxDOS.2 ?

**Comment by stsp** [comment]
Simply because I removed `loadaddress` so
the checks had to be updated.  I am not sure
they are either needed or valid. We used to load
"good" images, not some malicious ones, so
these checks will never trigger. I don't think they
are valid because they assume the certain magic
value 8192 - I'd say its WAY too much for a DOS
stack, plus freedos switches the stack at the start,
so even if it exceeded 8192 - the corruption won't
happen. I'd say something like 256 bytes would be
enough before the stack switch happens, not 8K.

If there is a need of some check (why though?),
its not a problem to get it back, but it should be
cleaner than before.
